### DISCUSSION TITLE
Using mavros  on red drone

### DISCUSSION THREAD

**Discussion by hinnatariq** [larics/icuas22_competition#66] [Q&A]
Hello @lmark1, @AnaBatinovic & @fpetric,

I know it's late to ask here but I have a question. I tried using move base navigation stack (http://wiki.ros.org/move_base)  with the red drone. But If we integrate it, it's not compatible as in the backend the drone is using ardupilot SITL. As far as I analyzed I'm not sure if we can use the ardupilot sitl and any existing exploration package (like frontier or RRT ) to do autonomous navigation without a prior map since they require velocity commands not positional data. And when I try to communicate with ROS with the red drone control architecture using mavros frame work, it doesn't seem to move at all, yet I don't see any errors. It seems like some backend algorithm is only responsible for arming and making it hover. Can you please give some explanation on this and any suggestions how to use mavros on red drone, may be any of your previous work or open-source github repos that can see and try as I am taking it as a hobby project. Thank you

**Comment by hinnatariq** [comment]
@lmark1 @AnaBatinovic @fpetric can you please give guide me?

**Comment by fpetric** [comment]
Hi, sorry for not replying sooner, we were busy with evaluations :) 

You do not need move base in our setup, you just forward poses/waypoints to topp-ros node and then the trajectory is generated and the UAV executes that.

Basically, any package that can generate a waypoint or a set of waypoints, you can just forward that set of waypoints to the topp-ros node, which will then generate trajectory and the trajectory tracker will take care of executing it. The trajectory needs to be published to `tracker/input_trajectory`. If you use a single waypoint, you can use `trajectory/input`_pose (as stated [here](https://github.com/larics/icuas22_competition#controlling-the-uav)).

Our setup works based on the position controller, and you can send a reference to that controller via `position_hold/trajectory`. If you want to emulate velocity control, you can just use the velocity reference from any of the packages you use and generate a waypoint in the direction of the velocity vector. With a bit of tuning of the position reference generator and controller I think you could achieve something similar to velocity control. 

And if you want to see an example of this exact setup in an exploration task, feel free to check out [nbvp_exploration package](https://github.com/larics/nbvp_exploration) and the accompanying [wiki page](https://github.com/larics/nbvp_exploration/wiki).
